pkg/kubelet/dockertools/manager.go

```diff
@@ -276,6 +276,38 @@ type containerStatusResult struct {
 	err    error
 }
 
+// determineInfraContainerIP inspects the given container and determines the IP address of that
+// container.  The container is assumed to be the pod infra container; it is the responsibility
+// of the caller to ensure that the container ID passed is that of the infra container.
+func (dm *DockerManager) determineInfraContainerIP(dockerID string, pod *api.Pod) (string, error) {
+	inspectResult, err := dm.client.InspectContainer(dockerID)
+	if err != nil {
+		return "", err
+	}
+	if inspectResult == nil {
+		return "", nil // TODO: should be an error, probably
```

Comment:
Write this code where you can tolerate not having the podIP, but you should
not ignore any errors in this function.

On Fri, Aug 21, 2015 at 4:42 PM, Paul Morie <notifications@github.com>
wrote:

> In pkg/kubelet/dockertools/manager.go
> <https://github.com/kubernetes/kubernetes/pull/13052#discussion_r37676100>
> :
>
> > @@ -276,6 +276,38 @@ type containerStatusResult struct {
> >  	err    error
> >  }
> >
> > +// determineInfraContainerIP inspects the given container and determines the IP address of that
> > +// container.  The container is assumed to be the pod infra container; it is the responsibility
> > +// of the caller to ensure that the container ID passed is that of the infra container.
> > +func (dm *DockerManager) determineInfraContainerIP(dockerID string, pod *api.Pod) (string, error) {
> > +	inspectResult, err := dm.client.InspectContainer(dockerID)
> > +	if err != nil {
> > +		return "", err
> > +	}
> > +	if inspectResult == nil {
> > +		return "", nil // TODO: should be an error, probably
>
> This if clause is bothering me; inspectContainer does it but it seems
> like it would represent a bug in the docker API to get an (empty
> container, nil) result from InspectContainer. It would be great if there
> were a comment on the original code with a reference to an issue either in
> kube or docker, but oh well. I'm strongly tempted to take this clause out...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13052/files#r37676100>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift